File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentUploadResponse(BaseModel):
    filename: str
    chunks_indexed: int
    message: str


class DocumentSummary(BaseModel):
    filename: str
    file_type: str
    size_kb: float
    updated_at: datetime
    status: str

class DocumentUpdateResponse(BaseModel):
    filename: str
    chunks_indexed: int
    message: str


class DocumentListResponse(BaseModel):
    documents: list[DocumentSummary]

File: frontend/src/app/models/document.ts
export interface DocumentUploadResponse {
  filename: string;
  chunks_indexed: number;
  message: string;
}

export interface DocumentSummary {
  filename: string;
  file_type: string;
  size_kb: number;
  updated_at: string;
  status: string;
}

export interface DocumentListResponse {
  documents: DocumentSummary[];
}